### ISSUE TITLE
Poor testing results

### ISSUE THREAD

**Issue by tejasri19** [cxliu0/PET#31]
I trained PET model on my custom crop dataset to count crop heads for 250 epochs. 

> [ep 241][lr 0.0001000][95.76s]{"train_lr": 9.999999999999967e-05, "train_loss": 0.06341644998685253, "train_loss_ce_sp": 0.038234214986697206, "train_loss_points_sp": 0.0022481676240781005, "train_loss_ce_ds": 0.022710887208671832, "train_loss_points_ds": 0.00022298315949539267, "train_loss_split": 1.9718961237636107e-07, "train_loss_ce_sp_unscaled": 0.038234214986697206, "train_loss_points_sp_unscaled": 0.00044963352482271855, "train_loss_ce_ds_unscaled": 0.022710887208671832, "train_loss_points_ds_unscaled": 4.4596631920374024e-05, "train_loss_split_unscaled": 1.971896101788777e-06, "epoch": 241, "n_parameters": 17882569}

[ep 242][lr 0.0001000][95.30s]{"train_lr": 9.999999999999967e-05, "train_loss": 0.06283839971690279, "train_loss_ce_sp": 0.037631992293271896, "train_loss_points_sp": 0.002375330376756809, "train_loss_ce_ds": 0.022599477666748188, "train_loss_points_ds": 0.00023117944152412437, "train_loss_split": 4.2016914054265614e-07, "train_loss_ce_sp_unscaled": 0.037631992293271896, "train_loss_points_sp_unscaled": 0.000475066075496881, "train_loss_ce_ds_unscaled": 0.022599477666748188, "train_loss_points_ds_unscaled": 4.623588839178144e-05, "train_loss_split_unscaled": 4.2016913251179025e-06, "epoch": 242, "n_parameters": 17882569}

[ep 243][lr 0.0001000][96.40s]{"train_lr": 9.999999999999967e-05, "train_loss": 0.0626774635245433, "train_loss_ce_sp": 0.03795098314379774, "train_loss_points_sp": 0.0020518471975242945, "train_loss_ce_ds": 0.022445354364505718, "train_loss_points_ds": 0.00022890057823643452, "train_loss_split": 3.781551349910975e-07, "train_loss_ce_sp_unscaled": 0.03795098314379774, "train_loss_points_sp_unscaled": 0.0004103694389014868, "train_loss_ce_ds_unscaled": 0.022445354364505718, "train_loss_points_ds_unscaled": 4.578011557452733e-05, "train_loss_split_unscaled": 3.7815512680425877e-06, "epoch": 243, "n_parameters": 17882569}

[ep 244][lr 0.0001000][96.04s]{"train_lr": 9.999999999999967e-05, "train_loss": 0.061275575314534876, "train_loss_ce_sp": 0.03655859408332262, "train_loss_points_sp": 0.002326323804388777, "train_loss_ce_ds": 0.022168093942469214, "train_loss_points_ds": 0.0002223615300353253, "train_loss_split": 2.0239411622647022e-07, "train_loss_ce_sp_unscaled": 0.03655859408332262, "train_loss_points_sp_unscaled": 0.0004652647601111179, "train_loss_ce_ds_unscaled": 0.022168093942469214, "train_loss_points_ds_unscaled": 4.4472305982220325e-05, "train_loss_split_unscaled": 2.023941133080459e-06, "epoch": 244, "n_parameters": 17882569}

[ep 245][lr 0.0001000][95.83s]{"train_lr": 9.999999999999967e-05, "train_loss": 0.06415569219374802, "train_loss_ce_sp": 0.038525093940836266, "train_loss_points_sp": 0.0023453568700394344, "train_loss_ce_ds": 0.0230697781778872, "train_loss_points_ds": 0.00021521263832521693, "train_loss_split": 2.504558136667698e-07, "train_loss_ce_sp_unscaled": 0.038525093940836266, "train_loss_points_sp_unscaled": 0.00046907137407177334, "train_loss_ce_ds_unscaled": 0.0230697781778872, "train_loss_points_ds_unscaled": 4.304252768456425e-05, "train_loss_split_unscaled": 2.504558098025438e-06, "epoch": 245, "n_parameters": 17882569}

epoch:245, mae:53.97073170731707, mse:75.53871564287819, time20.52969455718994, 

best mae:53.97073170731707, best epoch: 5

When I visualised, the results on test data are zero. My train:test data is 819:205. Each image resolution is 1533 x 1133.

@cxliu0 Please share your inputs

**Comment by tejasri19** [comment]
1. What should be the ideal number of epochs to be used?
2. What can be the learning rate to start with on the custom dataset?

**Comment by cxliu0** [comment]
Here are some tips to improve the results:

* Please make sure that the data format is correct. For example, you can visualize the training annotations to check whether the ground-truth points are correct.

* Based on the training loss, it appears that objects in your custom dataset are relatively sparse. You can use a smaller resolution to train the model, e.g., constraining the longer size of the image to 1024.

* You may use the default training epoch (i.e., 1500) to train the model.
